doc/compiler/optimizer/LoopOptimizationSummary.md

```diff
@@ -0,0 +1,114 @@
+<!--
+Copyright (c) 2022, 2022 IBM Corp. and others
+
+This program and the accompanying materials are made available under
+the terms of the Eclipse Public License 2.0 which accompanies this
+distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+or the Apache License, Version 2.0 which accompanies this distribution and
+is available at https://www.apache.org/licenses/LICENSE-2.0.
+
+This Source Code may also be made available under the following
+Secondary Licenses when the conditions for such availability set
+forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+General Public License, version 2 with the GNU Classpath
+Exception [1] and GNU General Public License, version 2 with the
+OpenJDK Assembly Exception [2].
+
+[1] https://www.gnu.org/software/classpath/license.html
+[2] http://openjdk.java.net/legal/assembly-exception.html
+
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# [Loop Optimizations](https://github.com/eclipse/omr/blob/master/doc/compiler/optimizer/IntroLoopOptimizations.md)
+
+## Loop Canonicalization
+Loop Canonicalization transforms `while` loops to if-guarded `do-while` loops
+with a loop invariant (pre-header) block. It places the loop test at the end
+of the trees for the loop, so that the loop back-edge is almost always a
+backwards branch.
+
+## Loop Versioning and Loop Specialization
+During Loop Versioning, loop invariant null checks, bound checks, div checks,
+checkcasts and inline guards (where the `this` does not change in the loop)
+are eliminated from loops by placing equivalent tests outside the loop. It
+also eliminates bound checks where the range of values of the index inside
+the loop is discovered (need to be a compile time constant) and checks are
+inserted outside the loop so that no bound checks fail inside the loop.
+
+This analysis uses induction variable information found by [Value Propagation](https://github.com/eclipse/omr/blob/master/doc/compiler/optimizer/ValuePropagation.md).
+Another versioning test that is emitted ensures the loop will not run for a
+very long period of time (in which case the async check inside the loop can
```

Comment:
Is there any place that describes what an async check?

Reply:
Not to my knowledge. Async checks get added by the VM at backward branches in the bytecodes on the assumption that one cannot have a loop without a backwards branch. The async check is used for many purposes all of which are some variant of the need to periodically have all application threads yield in order for some sort of action to be taken, with the prime examples being a) GC and b) walk the stack to sample where the thread is executing for driving compilation control. A better definition may be possible, but that hopefully gives the gist.